matchms/similarity/PrecursormzMatch.py

```diff
@@ -0,0 +1,164 @@
+        Precursor m/z match between 2 and 4 is 1.00
+
+    """
+    # Set key characteristics as class attributes
+    is_commutative = True
+
+    def __init__(self, tolerance: float = 0.1, tolerance_type: str = "Dalton"):
+        """
+        Parameters
+        ----------
+        tolerance
+            Specify tolerance below which two m/z are counted as match.
+        tolerance_type
+            Chose between fixed tolerance in Dalten (="Dalton") or a relative difference
+            in ppm (="ppm").
+        """
+        self.tolerance = tolerance
+        assert tolerance_type in ["Dalton", "ppm"], "Expected type from ['Dalton', 'ppm']"
+        self.type = tolerance_type
+
+    def pair(self, reference: SpectrumType, query: SpectrumType) -> float:
+        """Compare precursor m/z between reference and query spectrum.
+
+        Parameters
+        ----------
+        reference
+            Single reference spectrum.
+        query
+            Single query spectrum.
+        """
+        precursormz_ref = reference.get("precursor_mz")
+        precursormz_query = query.get("precursor_mz")
+        assert precursormz_ref is not None and precursormz_query is not None, "Missing precursor m/z."
+
+        if self.type == "Dalton":
+            return abs(precursormz_ref - precursormz_query) <= self.tolerance
+
+        mean_mz = (precursormz_ref + precursormz_query) / 2
+        return abs(precursormz_ref - precursormz_query)/mean_mz <= self.tolerance
+
+    def matrix(self, references: List[SpectrumType], queries: List[SpectrumType],
+               is_symmetric: bool = False) -> numpy.ndarray:
+        """Compare parent masses between all references and queries.
+
+        Parameters
+        ----------
+        references
+            List/array of reference spectrums.
+        queries
+            List/array of Single query spectrums.
+        is_symmetric
+            Set to True when *references* and *queries* are identical (as for instance for an all-vs-all
+            comparison). By using the fact that score[i,j] = score[j,i] the calculation will be about
+            2x faster.
+        """
+        def collect_precursormz(spectrums):
+            """Collect precursors."""
+            precursors = []
+            for spectrum in spectrums:
+                precursormz = spectrum.get("precursor_mz")
+                assert precursormz is not None, "Missing precursor m/z."
+                precursors.append(precursormz)
+            return numpy.asarray(precursors)
+
+        precursors_ref = collect_precursormz(references)
+        precursors_query = collect_precursormz(queries)
+        if is_symmetric and self.type == "Dalton":
+            return precursormz_scores_symmetric(precursors_ref, precursors_query, self.tolerance).astype(bool)
+        if is_symmetric and self.type == "ppm":
+            return precursormz_scores_symmetric_ppm(precursors_ref, precursors_query, self.tolerance).astype(bool)
```

Comment:
This line does not seem to be tested, see https://sonarcloud.io/code?id=matchms_matchms&pullRequest=156&selected=matchms_matchms%3Amatchms%2Fsimilarity%2FPrecursormzMatch.py&line=119

Can a test be added for it?

Reply:
I now added a test for this as well.

Reply:
great 100% coverage